rust/src/lib.rs

```diff
@@ -2679,6 +2784,107 @@ pub unsafe extern "C" fn zcashlc_shield_funds(
     unwrap_exc_or(res, false)
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn zcashlc_propose_shielding(
+    db_data: *const u8,
+    db_data_len: usize,
+    usk_ptr: *const u8,
+    usk_len: usize,
+    memo: *const u8,
+    shielding_threshold: u64,
+    network_id: u32,
+    min_confirmations: u32,
+    use_zip317_fees: bool,
+    proposal_bytes_ret: *mut u8
+) -> bool {
+    let res = catch_panic(|| {
+        let network = parse_network(network_id)?;
+        let min_confirmations = NonZeroU32::new(min_confirmations)
+            .ok_or(anyhow!("min_confirmations should be non-zero"))?;
+        let mut db_data = unsafe { wallet_db(db_data, db_data_len, network)? };
+
+        let usk = unsafe { decode_usk(usk_ptr, usk_len) }?;
+
+        let memo_bytes = if memo.is_null() {
+            MemoBytes::empty()
+        } else {
+            MemoBytes::from_bytes(unsafe { slice::from_raw_parts(memo, 512) })
+                .map_err(|e| anyhow!("Invalid MemoBytes: {}", e))?
+        };
+
+        let shielding_threshold = NonNegativeAmount::from_u64(shielding_threshold)
+            .map_err(|()| anyhow!("Invalid amount, out of range"))?;
+
+        let account = db_data
+            .get_account_for_ufvk(&usk.to_unified_full_viewing_key())?
+            .ok_or_else(|| anyhow!("Spending key not recognized."))?;
+
+        let taddrs: Vec<TransparentAddress> = db_data
+            .get_target_and_anchor_heights(NonZeroU32::MIN)
```

Comment:
This isn't really an anchor height; you should probably just use `chain_tip - min_confirmations`.

Reply:
I'm confused about this one. This is just what zcashls_shield_funds was doing. Are you saying instead of making the call to `db_data. get_target_and_anchor_heights` to just pass `chain_tip - min_confirmations` to db_data.get_transparent_balances()`?

Reply:
Yup! You'll notice that that implementation has changed: https://github.com/zcash/librustzcash/blob/236cd569ee4a824d98920d1a61b8cc2b90ceba1d/zcash_client_backend/src/data_api/wallet.rs#L505-L519

Reply:
You can also just use the `propose_shielding` function directly.